### PR TITLE
style(fscomponents): modalhalfscreen vertical position

### PR DESCRIPTION
**Pull request by DavidRhodesBB**
sets up `verticalPosition` props with 'top', 'middle', and 'bottom' as default. Testable in **Development > PSHalfModal** (EmailSignUp, with `verticalPosition={'middle'}`).  

Handles https://github.com/brandingbrand/flagship/issues/309

### CODE REVIEW

File: packages/fscomponents/src/components/ModalHalfScreen.tsx [hunk 1]
```diff
@@ -19,11 +20,11 @@ const styles = StyleSheet.create({
     bottom: -400
   }
 });
-
 export interface ModalHalfScreenProps {
   visible?: boolean;
   height?: number;
   onRequestClose: () => void;
+  verticalPosition?: string;
```
Comment (bweissbart) [inline packages/fscomponents/src/components/ModalHalfScreen.tsx]: This should be defined as a enum with values top, middle, bottom: https://www.typescriptlang.org/docs/handbook/enums.html

File: packages/fscomponents/src/components/ModalHalfScreen.tsx [hunk 2]
```diff
@@ -93,16 +94,34 @@ export class ModalHalfScreen extends PureComponent<ModalHalfScreenProps, ModalHa
   }
 
   render(): JSX.Element {
+    let calculatedPosition;
+    const screenHeight = Dimensions.get('window').height;
```
Comment (bweissbart) [inline packages/fscomponents/src/components/ModalHalfScreen.tsx]: This should be implemented as a Dimensions listener so we don't have to recalculate this value with every render: https://github.com/brandingbrand/flagship/blob/3bfde0951d4a460f26ae2e283ae0996bae8c0e0a/packages/pirateship/src/screens/Category.tsx#L69

Comment (DavidRhodesBB) [reply packages/fscomponents/src/components/ModalHalfScreen.tsx]: Good stuff, thanks for catching that!  Implemented in updated PR.

File: packages/fscomponents/src/components/ModalHalfScreen.tsx (hunk 1 above)
Comment (DavidRhodesBB) [reply packages/fscomponents/src/components/ModalHalfScreen.tsx]: I should have worked harder implementing that - Thanks for the learning on how to use enums @bweissbart.  I also added a small library for calculating the StatusBar height when using 'top' position.

**Pull request closed without merging**